README.md

```diff
@@ -19,25 +19,70 @@ yarn add --dev @netlify/ipx
 Create `netlify/functions/ipx.ts`:
 
 ```ts
-import { createIPXHandler } from '@netlify/ipx'
+import { createIPXHandler } from "@netlify/ipx";
 
 export const handler = createIPXHandler({
-  domains: ['images.unsplash.com']
-})
+  domains: ["images.unsplash.com"],
+});
 ```
 
 Now you can use IPX to optimize both local and remote assets ✨
 
 Resize `/test.jpg` (in `dist`):
 
 ```html
-<img src="/.netlify/functions/ipx/w_200/static/test.jpg"/>
+<img src="/.netlify/functions/ipx/w_200/static/test.jpg" />
 ```
 
 Resize and change format for a remote url:
 
 ```html
-<img src="/.netlify/functions/ipx/f_webp,w_450/https://images.unsplash.com/photo-1514888286974-6c03e2ca1dba"/>
+<img
+  src="/.netlify/functions/ipx/f_webp,w_450/https://images.unsplash.com/photo-1514888286974-6c03e2ca1dba"
+/>
+```
+
+## Remote Patterns
```

Comment:
I am wondering whether implementing it exactly as `remotePatterns` is tying ourselves too tightly to Next.js. I kinda feel like we should just have it as URLs with wildcards allowed. That would be easier to write, and not hard to generate at the Next.js end. So, it could be:

```json
{
remotePatterns: ['https://example.com:3000/blog/**']
}
```

Reply:
You could even load them in a new `URL` to get all the pieces and types out of the box.

```
> new URL('https://example.com:3000/blog/**')
URL {
  href: 'https://example.com:3000/blog/**',
  origin: 'https://example.com:3000',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'example.com:3000',
  hostname: 'example.com',
  port: '3000',
  pathname: '/blog/**',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

Reply:
@matthojo I kept it the same as Next's implementation as only `hostname` is required in RemotePattern, so the conditional checks on protocol, port, and pathname that I have in util.ts only happen if a user has specified that they want to check on those things. If a user is only specifying the `hostname` in their next config, then we would need to make assumptions on the protocol (and the pathname as `**`) which I'm not sure we want to make?